AllisonsCatPalaceTDDTests/LocationControllerTests.swift

```diff
@@ -0,0 +1,332 @@
+
+        guard let handler = geocoder.forwardGeocodeAddressCompletionHandler else {
+            return XCTFail("Geocoder should be called with a handler")
+        }
+        handler([], GeocodingError.noLocationsFound)
+
+        XCTAssertEqual(textField.layer.borderColor, UIColor.red.cgColor,
+                       "Zip code field should have red border on error")
+        // TODO: maybe show an error message depending on the type of error that comes back?
+    }
+
+    func testGeocodingWithEmptyResultsAndNoError() {
+        delegate.textFieldDidEndEditing!(textField)
+
+        guard let handler = geocoder.forwardGeocodeAddressCompletionHandler else {
+            return XCTFail("Geocoder should be called with a handler")
+        }
+        handler([], nil)
+
+        XCTAssertEqual(textField.layer.borderColor, UIColor.red.cgColor,
+                       "Zip code field should have red border when there are no results")
+        // TODO: use custom error for empty results to show the message
+    }
+
+    func testGeocodingWithNonEmptyResultsAndError() {
+        delegate.textFieldDidEndEditing!(textField)
+
+        guard let handler = geocoder.forwardGeocodeAddressCompletionHandler else {
+            return XCTFail("Geocoder should be called with a handler")
+        }
+        handler([placemark], GeocodingError.noLocationsFound)
+
+        XCTAssertEqual(textField.layer.borderColor, UIColor.red.cgColor,
+                       "Zip code field should have red border when there is an error regardless of results")
+        // TODO: show the error? Ignore the error if you have the info you want already?
+    }
+
+    func testGeocodingWithNonEmptyResultsAndNoError() {
+        delegate.textFieldDidEndEditing!(textField)
+
+        guard let handler = geocoder.forwardGeocodeAddressCompletionHandler else {
+            return XCTFail("Geocoder should be called with a handler")
+        }
+        handler([placemark], nil)
+
+        XCTAssertEqual(textField.layer.borderColor, UIColor.lightGray.cgColor,
+                       "Zip code field should have non-error color border when there are results and no error")
+    }
+
+    func testSuccessfulGeocodingShowsList() {
+        let navController = UIStoryboard(name: "Main", bundle: nil).instantiateInitialViewController() as! UINavigationController
+        navController.addChildViewController(controller)
+
+        let performSegueSpy = UIViewController.PerformSegueSpyController.createSpy(on: controller)
+        let showSpy = UIViewController.ShowSpyController.createSpy(on: navController)
+
+        performSegueSpy?.beginSpying()
+        showSpy?.beginSpying()
+
+        delegate.textFieldDidEndEditing!(textField)
+
+        guard let handler = geocoder.forwardGeocodeAddressCompletionHandler else {
+            return XCTFail("Geocoder should be called with a handler")
+        }
+        handler([placemark], nil)
+
+        XCTAssertTrue(navController.showCalled,
```

Comment:
Test the segue by finding it in the storyboard and then test that it's a show and that it has the correct destination.